clib/cMap.mli

```diff
@@ -111,6 +111,9 @@ module type ExtS = sig
 
   val fold_right : (key -> 'a -> 'b -> 'b) -> 'a t -> 'b -> 'b
   (** Folding keys in decreasing order. *)
+
+  val fold_left_map : (key -> 'a -> 'b -> 'a * 'c) -> 'a -> 'b t -> 'a * 'c t
+  (** map with a side effect *)
```

Comment:
Maybe "combination of fold_left and map". I understand what "side effect" means here, but it may sounds like some imperative side effect, whereas the interface remains purely functional.